src/main/java/com/geoxp/GeoXPLib.java

```diff
@@ -58,6 +58,23 @@
   }
   public static final class GeoXPShape implements Serializable {
     long[] geocells;
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      GeoXPShape that = (GeoXPShape) o;
+      return Arrays.equals(geocells, that.geocells);
```

Comment:
Since you are comparing the array content maybe sorting the geocells prior to comparing them would be better?

Reply:
Second thoughts though, adding a systematic sorting of the geocells could greatly impact performance, so maybe when computing the hashcode, if it differs from the previously computed hashcode then sort the array, and in equals, if the hashcodes differ return false, otherwise, compare the arrays using Array.equals?

Reply:
Aren't the geocells always sorted? `Coverage#contains` certainly expects the array to be sorted as it uses `Arrays.binarysSearch`.
I'm not sure about duplicates geocells, but I suspect there aren't any.

Reply:
Technically you could create an instance of GeoXPShape from an unsorted / redundant long array. Let's leave it this way for now.